src/utils.jl

```diff
@@ -258,26 +258,74 @@ end
 
 # These getters improve inference since fieldtype(CodeInfo, :linetable)
 # and fieldtype(CodeInfo, :codelocs) are both Any
-const LineTypes = Union{LineNumberNode,Core.LineInfoNode}
+@static if VERSION ≥ v"1.12.0-DEV.173"
+    const LineTypes = Base.IRShow.LineInfoNode
+else
+    const LineTypes = Union{LineNumberNode,Core.LineInfoNode}
+end
 function linetable(arg)
     if isa(arg, Frame)
         arg = arg.framecode
     end
     if isa(arg, FrameCode)
         arg = arg.src
     end
-    return (arg::CodeInfo).linetable::Union{Vector{Core.LineInfoNode},Vector{Any}}  # issue #264
+    ci = arg::CodeInfo
+    @static if VERSION ≥ v"1.12.0-DEV.173"
+        return ci.debuginfo
+    else
+        return ci.linetable::Union{Vector{Core.LineInfoNode},Vector{Any}} # issue #264
+    end
 end
 _linetable(list::Vector, i::Integer) = list[i]::Union{Expr,LineTypes}
 function linetable(arg, i::Integer; macro_caller::Bool=false)::Union{Expr,LineTypes}
     lt = linetable(arg)
-    lineinfo = _linetable(lt, i)
-    if macro_caller
-        while lineinfo isa Core.LineInfoNode && lineinfo.method === Symbol("macro expansion") && lineinfo.inlined_at != 0
-            lineinfo = _linetable(lt, lineinfo.inlined_at)
+    @static if VERSION ≥ v"1.12.0-DEV.173"
+        # TODO: decode the linetable at this frame efficiently by reimplementing this here
+        # TODO: get the contextual name from the parent, rather than returning "n/a" (which breaks Cthulhu)
+        return Base.IRShow.buildLineInfoNode(lt, :var"n/a", i)[1] # ignore all inlining / macro expansion / etc :(
```

Comment:
I don't understand this commit -- do we now store some macro expansion context within `debuginfo`?